tests/test_client.py

```diff
@@ -110,3 +110,38 @@ def test_client_invalid_auth_token(mocker, mock_authorizer_tuple, token, mock_ke
     # Call the Garden constructor and expect an auth exception
     with pytest.raises(AuthException):
         GardenClient(auth_client=mock_auth_client)
+
+
+def test_local_storage_garden(mocker, garden_client, garden_all_fields, tmp_path):
+    # mock to replace "~/.garden/db"
+    mocker.patch("garden_ai.client.LOCAL_STORAGE", new=tmp_path)
+    uuid = garden_all_fields.uuid
+    garden_client.put_local(garden_all_fields)
+    record = garden_client.get_local(uuid)
+    assert record == garden_all_fields.json()
+
+
+def test_local_storage_pipeline(mocker, garden_client, pipeline_toy_example, tmp_path):
+    # mock to replace "~/.garden/db"
+    mocker.patch("garden_ai.client.LOCAL_STORAGE", new=tmp_path)
+    uuid = pipeline_toy_example.uuid
+    garden_client.put_local(pipeline_toy_example)
+    record = garden_client.get_local(uuid)
+    assert record == pipeline_toy_example.json()
+
+
+def test_local_storage_keyerror(mocker, garden_client, garden_all_fields, tmp_path):
+    # mock to replace "~/.garden/db"
+    tmp_path.mkdir(parents=True, exist_ok=True)
+    mocker.patch("garden_ai.client.LOCAL_STORAGE", new=tmp_path)
```

Comment:
Thinking out loud ... it would be great if our tests weren't doing a lot of filesystem mocking.  Probably not worth rethinking it on this PR, but I think we can get to a place where our tests are just manipulating a dict in memory and we can mock out the `read_from_disk` and `write_to_disk` calls.